README.md
# wfrp-chargen
Automatically exported from 

WFRP-CharGen?
Character Generator/tracking for Warhammer Fantasy Role Play V2

Goals
The goal is a cross platform applications that will allow you to generate and maintain your Warhammer FRP characters.

Status
In development. It currently works and not all the rules are followed. 
See the Screen Shots for a glimps at the project.

To help with the project, please join the google group. The link is on the right hand side.

Current Topics
The Current topics of discussion are:

Trappings
Change Log
History
Started in 2005 and written in java, this generator was a to be the next evolution to a pc generator written for 
the web (in php - See link on the side bar). With a couple months of development, the project was shelved. Now, 
the start of March 2009, the project has arisen again. Instead of using flat text files, XML was chosen as the 
back-end source for races and careers.

Disclaimer
Warhammer and other Warhammer Fantasy Role Play (and the Logo's) are (probably registered) trademarks of Games 
Workshop and/or Green Ronin and/or Black Industries and/or Fantasy Flight Games. The use of trademarks and materials 
are not meant as a challenge to their rights and is not intended to make or lose any money for anyone.



ChangeLog.md
# Known Issues/Change Log #

## 9/13/2009 ##
  * Dwarf and Elf names are checked in with the new format.

## 9/06/2009 ##
  * Names are no longer dependent on the gender in the file name.
  * Names support multi elements, like Element1+Eelement2 equals a firstname.

## 9/05/2009 ##
  * fixed case issue with datafiles vs dataFiles ([bug 14](https://code.google.com/p/wfrp-chargen/issues/detail?id=4))

## 6/21/2009 ##
[r38](https://code.google.com/p/wfrp-chargen/source/detail?r=38) released - http://code.google.com/p/wfrp-chargen/downloads/list
  * editing panel now contains a way to enter experience points
  * default character is now set back to 0 (from 50000) since you can edit it now
  * compiled using java 1.5 since java 6 is not available for 32bit Macs

## 6/20/2009 ##
[r36](https://code.google.com/p/wfrp-chargen/source/detail?r=36) release - http://code.google.com/p/wfrp-chargen/downloads/list
  * editing characteristics now displays correctly when save is pressed.
  * editing is disabled until a pc is generated

## 6/18/2009 ##
  * When all advances are taken, it now says "None" instead of "Select Attribute".
  * removed debug output when birthplace is determined.
  * cleaned up some error checking output.
  * edit screen for characteristics - now displays all the edit-able stuff
> > - new Bug - saved data does not display until something else is selected

## 6/10/2009 ##
  * cleaned up the about screen
  * added an edit button
  * dropdown list for skills/talents no longer resize depending
> > on the length of the item.  Mouseovers show what is all inside
  * a character now gets the skills/talents of the first career
  * bug - where if you select "none" or "select skill" - they get added to
> > the skills/talents/profile
  * added exp tracking left and total used
> > - character XML has been modified - old characters will not work
> > unless  

&lt;ExpUsed&gt;

SOME AMOUNT

&lt;/ExpUsed&gt;

 is added to the 

&lt;Character&gt;

 section
> > - starting characters with 50,000 exp until something is written to
> > allow the player to edit the exp field.

## 5/24/2009 ##
Typo in Demagogue career fixed

Skills/Talents now check inside an OR for a matching skill and does not display if you have 3 of the same Skill or the talent already - [Issue 6](https://code.google.com/p/wfrp-chargen/issues/detail?id=6)

Started the About button


## 5/23/2009 ##
[r15](https://code.google.com/p/wfrp-chargen/source/detail?r=15) release - http://code.google.com/p/wfrp-chargen/downloads/list

Changes:
Save/Load now work

Printing has been stubbed out

> - printing prints to the console window

Shallya's Mercy has been added

Talents/Skills are now sorted when a new one is added.
Talents/Skills now stack correctly

"ANY" in skills now works - needs a little more testing but the first
few runs seemed to be ok


## 5/5/2009 ##

  * Wounds/Fate are not caculated - Issue number 3 - Fixed Checkin 17 on 5/5/2009

wfrp-chargen/src/wfrpv2/helpers/save.java
/*
 * Created on May 10, 2005
 */
package wfrpv2.helpers;
import rpg.XMLFile.XMLDocument;
import wfrpv2.dataTypes.Character;
import misc.intsStrings;

/**
 * @author Mathew Anderson
 * @author http://www.snotling.org
 *
 * Saves the character to an XML file.
 */
public class save {

	public static void main(Character character) throws Exception {
		XMLDocument XMLCharacter = new XMLDocument("Character");
		
		XMLCharacter.setFileName(character.name);
		
		XMLCharacter.setValue("/Character/Name", character.name);
		XMLCharacter.setValue("/Character/Race", character.race);
		XMLCharacter.setValue("/Character/Career", character.career);
		// Career path
		for ( int i=0; i<character.career_path.size(); i++ ) {
			String element = (String) character.career_path.get(i);
			System.out.print("\nElement "+ element);
			XMLCharacter.addValue("/Character", "career_path", element);
		}
		XMLCharacter.setValue("/Character/Gender", character.gender);
		XMLCharacter.setValue("/Character/Exp", intsStrings.toString(character.exp));
		XMLCharacter.setValue("/Character/ExpUsed", intsStrings.toString(character.expused));
		XMLCharacter.setValue("/Character/Height", intsStrings.toString(character.height));
		XMLCharacter.setValue("/Character/Weight", intsStrings.toString(character.weight));
		XMLCharacter.setValue("/Character/Age", intsStrings.toString(character.age));
		XMLCharacter.setValue("/Character/Haircolor", character.haircolor);
		XMLCharacter.setValue("/Character/Hairtype", character.hairtype);
		XMLCharacter.setValue("/Character/Eyecolor", character.eyecolor);
		XMLCharacter.setValue("/Character/Birthplace", character.birthplace);
		XMLCharacter.setValue("/Character/Starsign", character.starsign);
		XMLCharacter.setValue("/Character/Marks", character.marks);
		XMLCharacter.setValue("/Character/Siblings", intsStrings.toString(character.siblings));
		
		// Skills - talents -trappings
		for ( int i=0; i<character.skills.size(); i++ ) {
			String element = (String) character.skills.get(i);
			XMLCharacter.addValue("/Character", "skills", element);
		}
		for ( int i=0; i<character.talents.size(); i++ ) {
			String element = (String) character.talents.get(i);
				XMLCharacter.addValue("/Character", "talents", element);
		}
		for ( int i=0; i<character.trappings.size(); i++ ) {
			String element = (String) character.trappings.get(i);
			XMLCharacter.addValue("/Character", "trappings", element);
		}
		
		for ( int i=0; i<character.available_skills.size(); i++ ) {
			String element = (String) character.available_skills.get(i);
			XMLCharacter.addValue("/Character", "available_skills", element);
		}
		for ( int i=0; i<character.available_talents.size(); i++ ) {
			String element = (String) character.available_talents.get(i);
			XMLCharacter.addValue("/Character", "available_talents", element);
		}
		
		// And now for the profiles
		XMLCharacter.setValue("/Character/starting_profile/WS", intsStrings.toString(character.starting_profile[0]));
		XMLCharacter.setValue("/Character/starting_profile/BS", intsStrings.toString(character.starting_profile[1]));
		XMLCharacter.setValue("/Character/starting_profile/S", intsStrings.toString(character.starting_profile[2]));
		XMLCharacter.setValue("/Character/starting_profile/T", intsStrings.toString(character.starting_profile[3]));
		XMLCharacter.setValue("/Character/starting_profile/Ag", intsStrings.toString(character.starting_profile[4]));
		XMLCharacter.setValue("/Character/starting_profile/Int", intsStrings.toString(character.starting_profile[5]));
		XMLCharacter.setValue("/Character/starting_profile/WP", intsStrings.toString(character.starting_profile[6]));
		XMLCharacter.setValue("/Character/starting_profile/Fel", intsStrings.toString(character.starting_profile[7]));
		XMLCharacter.setValue("/Character/starting_profile/A", intsStrings.toString(character.starting_profile[8]));
		XMLCharacter.setValue("/Character/starting_profile/W", intsStrings.toString(character.starting_profile[9]));
		XMLCharacter.setValue("/Character/starting_profile/SB", intsStrings.toString(character.starting_profile[10]));
		XMLCharacter.setValue("/Character/starting_profile/TB", intsStrings.toString(character.starting_profile[11]));
		XMLCharacter.setValue("/Character/starting_profile/M", intsStrings.toString(character.starting_profile[12]));
		XMLCharacter.setValue("/Character/starting_profile/Mag", intsStrings.toString(character.starting_profile[13]));
		XMLCharacter.setValue("/Character/starting_profile/IP", intsStrings.toString(character.starting_profile[14]));
		XMLCharacter.setValue("/Character/starting_profile/FP", intsStrings.toString(character.starting_profile[15]));
		
		XMLCharacter.setValue("/Character/advance_scheme/WS", intsStrings.toString(character.advance_scheme[0]));
		XMLCharacter.setValue("/Character/advance_scheme/BS", intsStrings.toString(character.advance_scheme[1]));
		XMLCharacter.setValue("/Character/advance_scheme/S", intsStrings.toString(character.advance_scheme[2]));
		XMLCharacter.setValue("/Character/advance_scheme/T", intsStrings.toString(character.advance_scheme[3]));
		XMLCharacter.setValue("/Character/advance_scheme/Ag", intsStrings.toString(character.advance_scheme[4]));
		XMLCharacter.setValue("/Character/advance_scheme/Int", intsStrings.toString(character.advance_scheme[5]));
		XMLCharacter.setValue("/Character/advance_scheme/WP", intsStrings.toString(character.advance_scheme[6]));
		XMLCharacter.setValue("/Character/advance_scheme/Fel", intsStrings.toString(character.advance_scheme[7]));
		XMLCharacter.setValue("/Character/advance_scheme/A", intsStrings.toString(character.advance_scheme[8]));
		XMLCharacter.setValue("/Character/advance_scheme/W", intsStrings.toString(character.advance_scheme[9]));
		XMLCharacter.setValue("/Character/advance_scheme/SB", intsStrings.toString(character.advance_scheme[10]));
		XMLCharacter.setValue("/Character/advance_scheme/TB", intsStrings.toString(character.advance_scheme[11]));
		XMLCharacter.setValue("/Character/advance_scheme/M", intsStrings.toString(character.advance_scheme[12]));
		XMLCharacter.setValue("/Character/advance_scheme/Mag", intsStrings.toString(character.advance_scheme[13]));
		XMLCharacter.setValue("/Character/advance_scheme/IP", intsStrings.toString(character.advance_scheme[14]));
		XMLCharacter.setValue("/Character/advance_scheme/FP", intsStrings.toString(character.advance_scheme[15]));
		
		XMLCharacter.setValue("/Character/advance_taken/WS", intsStrings.toString(character.advance_taken[0]));
		XMLCharacter.setValue("/Character/advance_taken/BS", intsStrings.toString(character.advance_taken[1]));
		XMLCharacter.setValue("/Character/advance_taken/S", intsStrings.toString(character.advance_taken[2]));
		XMLCharacter.setValue("/Character/advance_taken/T", intsStrings.toString(character.advance_taken[3]));
		XMLCharacter.setValue("/Character/advance_taken/Ag", intsStrings.toString(character.advance_taken[4]));
		XMLCharacter.setValue("/Character/advance_taken/Int", intsStrings.toString(character.advance_taken[5]));
		XMLCharacter.setValue("/Character/advance_taken/WP", intsStrings.toString(character.advance_taken[6]));
		XMLCharacter.setValue("/Character/advance_taken/Fel", intsStrings.toString(character.advance_taken[7]));
		XMLCharacter.setValue("/Character/advance_taken/A", intsStrings.toString(character.advance_taken[8]));
		XMLCharacter.setValue("/Character/advance_taken/W", intsStrings.toString(character.advance_taken[9]));
		XMLCharacter.setValue("/Character/advance_taken/SB", intsStrings.toString(character.advance_taken[10]));
		XMLCharacter.setValue("/Character/advance_taken/TB", intsStrings.toString(character.advance_taken[11]));
		XMLCharacter.setValue("/Character/advance_taken/M", intsStrings.toString(character.advance_taken[12]));
		XMLCharacter.setValue("/Character/advance_taken/Mag", intsStrings.toString(character.advance_taken[13]));
		XMLCharacter.setValue("/Character/advance_taken/IP", intsStrings.toString(character.advance_taken[14]));
		XMLCharacter.setValue("/Character/advance_taken/FP", intsStrings.toString(character.advance_taken[15]));
		
		XMLCharacter.setValue("/Character/current_profile/WS", intsStrings.toString(character.current_profile[0]));
		XMLCharacter.setValue("/Character/current_profile/BS", intsStrings.toString(character.current_profile[1]));
		XMLCharacter.setValue("/Character/current_profile/S", intsStrings.toString(character.current_profile[2]));
		XMLCharacter.setValue("/Character/current_profile/T", intsStrings.toString(character.current_profile[3]));
		XMLCharacter.setValue("/Character/current_profile/Ag", intsStrings.toString(character.current_profile[4]));
		XMLCharacter.setValue("/Character/current_profile/Int", intsStrings.toString(character.current_profile[5]));
		XMLCharacter.setValue("/Character/current_profile/WP", intsStrings.toString(character.current_profile[6]));
		XMLCharacter.setValue("/Character/current_profile/Fel", intsStrings.toString(character.current_profile[7]));
		XMLCharacter.setValue("/Character/current_profile/A", intsStrings.toString(character.current_profile[8]));
		XMLCharacter.setValue("/Character/current_profile/W", intsStrings.toString(character.current_profile[9]));
		XMLCharacter.setValue("/Character/current_profile/SB", intsStrings.toString(character.current_profile[10]));
		XMLCharacter.setValue("/Character/current_profile/TB", intsStrings.toString(character.current_profile[11]));
		XMLCharacter.setValue("/Character/current_profile/M", intsStrings.toString(character.current_profile[12]));
		XMLCharacter.setValue("/Character/current_profile/Mag", intsStrings.toString(character.current_profile[13]));
		XMLCharacter.setValue("/Character/current_profile/IP", intsStrings.toString(character.current_profile[14]));
		XMLCharacter.setValue("/Character/current_profile/FP", intsStrings.toString(character.current_profile[15]));
			
		// Now that the xml datatype is loaded.  save the file out.
		XMLCharacter.writeFile();
	}
}